### PR TITLE
[PW-4449] Market pay notification handler 

### PR DESCRIPTION
**Pull request by AlexandrosMor**
## Summary

In this pull request we added the marketpay notification models and the notification handler which is responsible to deserialize the json notifications
## Tested scenarios

Created mocking unit tests  

**Fixed issue**:

### CODE REVIEW

**Review by msilvagarcia — APPROVED** [review]

File: Adyen/Util/MarketPayNotificationConverter.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                case "REPORT_AVAILABLE":
+                    notification = new ReportAvailableNotification();
+                    break;
+                case "TRANSFER_FUNDS":
+                    notification = new TransferFundsNotification();
+                    break;
+                case "DIRECT_DEBIT_INITIATED":
+                    notification = new DirectDebitInitiatedNotification();
+                    break;
+                case "PAYOUT_CONFIRMED":
```
Comment (Aleffio) [inline Adyen/Util/MarketPayNotificationConverter.cs:106]: can't find this particular one on specs, I think it has been removed

File: Adyen/Adyen.csproj [hunk 2]
```diff
@@ -22,7 +22,7 @@
     <PackageReleaseNotes></PackageReleaseNotes>
     <PackageIconUrl>https://www.adyen.com/dam/jcr:39adc4db-76f7-4b9e-aec6-bfa21db4153c/adyen-logo-green.png</PackageIconUrl>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <SignAssembly>true</SignAssembly>
+    <SignAssembly>false</SignAssembly>
```
[the remark this answers is not in the document]
Comment (AlexandrosMor) [reply Adyen/Adyen.csproj:25]: In order to debug it should be false and true when we create the nuget package so it is `strong named`.

File: Adyen.Test/MarketPayTests/MarketPayNotificationTest.cs [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+            IGenericNotification notificationMessage = notificationHandler.HandleMarketpayNotificationJson(json);
+            AccountCloseNotification accountCloseNotification = (AccountCloseNotification)notificationMessage;
+            Assert.AreEqual(accountCloseNotification.EventType, "ACCOUNT_CLOSED");
+            Assert.AreEqual(accountCloseNotification.PspReference, "TSTPSPR0001");
+            Assert.AreEqual(accountCloseNotification.Content.ResultCode, "Success");
+        }
+
+
+
+        //    [TestMethod]
```
[the remark this answers is not in the document]
Comment (AlexandrosMor) [reply Adyen.Test/MarketPayTests/MarketPayNotificationTest.cs]: yeap fixed, good catch !

File: Adyen/Model/MarketPay/Notification/AccountFundsBelowThresholdNotificationContent.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Adyen.Model.MarketPay.Notification
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    [DataContract]
+    public class AccountFundsBelowThresholdNotificationContent : IGenericNotification
```
[the remark this answers is not in the document]
Comment (AlexandrosMor) [reply Adyen/Model/MarketPay/Notification/AccountFundsBelowThresholdNotificationContent.cs]: The interface was removed

File: Adyen/Util/MarketPayNotificationConverter.cs (hunk 1 above)
Comment (AlexandrosMor) [reply Adyen/Util/MarketPayNotificationConverter.cs]: PAYOUT_CONFIRMED removed based on MarketPayNotificationService-v7.json

**Review by msilvagarcia — APPROVED** [review]

**Review by Aleffio — APPROVED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Pull request merged**